### PR TITLE
feat(core) global plugins

### PR DESCRIPTION
**Pull request by thefosk**
See #1369.

### CODE REVIEW

**Comment by thefosk** [comment]
@thibaultCha @Tieske any feedback before merging?

File: spec/02-integration/05-proxy/03-plugins_triggering_spec.lua [hunk 1]
```diff
@@ -35,27 +77,52 @@ describe("Plugins triggering", function()
 
   teardown(function()
     if client then client:close() end
-    assert(helpers.stop_kong())
+    helpers.kill_all()
```
Comment (thibaultCha) [inline spec/02-integration/05-proxy/03-plugins_triggering_spec.lua]: `stop_kong` should be just fine here, why changing it?

**Comment by thibaultCha** [comment]
As always, would be helpful to properly explain which parts of #295 and #505 are implemented and how before having to figure it out by reading the patch...

**Comment by BrunoQuaresma** [comment]
Is there a doc about this feature? The only way that I saw on docs was enabling the plugin for each API.

**Pull request merged**